crates/wasmtime/src/engine.rs

```diff
@@ -530,6 +534,14 @@ impl Engine {
             flag
         ))
     }
+
+    pub(crate) fn is_branch_protection_enabled(
+        &self,
+        isa_flags: &BTreeMap<String, FlagValue>,
```

Comment:
Given that `isa_flags` always come from the `Engine` itself could this parameter be dropped and internally this uses `self.compiler().isa_flags()`?

Reply:
Also if it's a perf concern it might be prudent to add `engine.compiler().is_branch_protection_enabled(..)` since `isa_flags()` returns a whole new `BTreeMap` each time it's called I think. Or otherwise watch it get enabled on `Config` and ferry through that `bool` to here or similar.

Reply:
There is one case in which the flags come from a previously compiled module - [line 230](https://github.com/bytecodealliance/wasmtime/pull/3693/files#diff-f2a833f03cb2f0dfcd4dfda02740e498dea0ce15fc9749d67c711999c22ea5cfR230) in `crates/wasmtime/src/module/serialization.rs`. However, I can create a wrapper function, say `is_branch_protection_enabled_globally()`, that simply calls this one with a fixed parameter, `&engine.compiler().isa_flags()`, and change most call sites to invoke it instead.